articles/quickstart/webapp/nodejs-ea/01-login.md

```diff
@@ -0,0 +1,151 @@
+- `express-openid-connect` - an [OIDC](/protocols/oidc) compliant authentication middleware.
+
+::: note
+To learn more about using middleware with Express view the [official documentation](https://expressjs.com/en/guide/using-middleware.html)
+:::
+
+::: note
+Secrets should never be checked into source control.  We recommend using environment variables in place of plain text.  You can use [Dotenv](https://github.com/motdotla/dotenv) to easily manage envrionment variables on your localhost.
+:::
+
+```js
+const express = require('express');
+const session = require('express-session');
+const { auth } = require('express-openid-connect');
+
+// parses urlencoded payloads sent back from the Auth0 server
+app.use(express.urlencoded({
+  extended: false
+}));
+
+// session middleware required to handle user sessions
+app.use(session({
+  secret: 'this should be a secret',
+  resave: true,
+  saveUninitialized: false
+}));
+
+// express-openid-connect.auth 
+app.use(auth({
+  // required - forces authentication for all routes
+  required: false,
+  // baseURL - There is where your /callback, /login & /logout URLs will be registered
+  baseURL: 'http://localhost:3000',
+  // issuerBaseURL - This is your Auth0 Domain found in your Auth0 Application Settings
+  issuerBaseURL: '${account.namespace}',
+  // clientID - This is your Auth0 Client ID found in your Auth0 Application Settings
+  clientID: '${account.clientId}',
+  // clientSecret - This is your Auth0 Client Secret found in your Auth0 Application Settings
+  clientSecret: '${account.clientSecret}'
+}));
+```
+
+## Protect a Route
+Now that your application has the required middleware to integrate Auth0 in your application, you can start protecting routes.
+
+Add the `requiresAuth` middleware supplied from `express-openid-connect` for routes that need protection.  Any time a route using this middleware is hit, it will check for a valid user session, if none exists it will ask the user to log in.
+
+```js
+const { requiresAuth } = require('express-openid-connect');
+
+app.get('/logged_in_user', requiresAuth(), (req, res) => {
+  // when a user is logged in you can access the user object via `req.openid.user`
+  res.send(JSON.stringify(req.openid.user));
+});
+```
+
+## Login and Logout
+### Login
+To log a user in you have 3 options:
+- Send a `GET` request to the `/login` route registered by the `express-oidc-connect.auth` middleware.
+- Use the `express-oidc-connect.requiresAuth` middleware to protect a route.
+- When initializing the `express-oidc-connect.auth` middleware pass in `required: true` to force required authentication on all routes.
+
+::: note
+If you are testing your application locally, you can login by visiting `http://localhost:3000/login`
+:::
+
+### Logged in Users
+Once a user is logged in it is attached to the `request` via the `req.openid` object. You can access the full user object via `req.openid.user`
```

Comment:
🤔 ... I'd reword this a bit. Using `request` here is confusing. Maybe something like:

> The user's profile is always stored in `req.openid.user`

This makes me wonder ... should we add some kind of helper like `req.openid.isAuthenticated()` to the SDK that returns boolean based on if `user` is empty or not?

Reply:
I wonder if this section was even needed, it is demonstrated how to access the user object in the protect a route section

Reply:
You could be right. 

Also ... the method I mentioned already exists :)

https://github.com/auth0/express-openid-connect/blob/master/lib/context.js#L26-L28